src/server/handlers/launchpad.js

```diff
@@ -545,9 +561,46 @@ export const findSnaps = async (req, res) => {
       );
       return { ...snap, snapcraft_data: snapcraftData };
     }));
-    if (req.session.user) {
-      await initializeMetrics(req.session.user.id, snaps);
-    }
+    await db.transaction(async (trx) => {
```

Comment:
I wonder if it would make sense to extract metrics/db related code our of API response handlers.

In this particular example the 'keep the database up to date' code, which is not directly related to 'findSnap' functionality takes half of the code of this method making it quite hard to read.

It seems that likely we will be adding more and more metrics related code in different places, so it may be a good point to start thinking of some good convention to do that.